packages/flutter_tools/lib/src/ios/ios_deploy.dart

```diff
@@ -311,14 +314,12 @@ class IOSDeployDebugger {
         }
         if (line.contains('PROCESS_STOPPED') ||
             line.contains('PROCESS_EXITED') ||
-            _lldbProcessExit.hasMatch(line)) {
-          // The app exited or crashed, so stop echoing the output.
-          // Don't pass any further ios-deploy debugging messages to the log reader after it exits.
-          _debuggerState = _IOSDeployDebuggerState.detached;
+            _lldbProcessExit.hasMatch(line) ||
+            _lldbProcessStopped.hasMatch(line)) {
+          // The app exited or crashed, so exit. Continue passing debugging
+          // messages to the log reader until it exits to capture crash dumps.
           _logger.printTrace(line);
-          if (!debuggerCompleter.isCompleted) {
-            debuggerCompleter.complete(false);
```

Comment:
This is now completed after the process exits:
https://github.com/flutter/flutter/pull/68844/files#diff-3f457a69ca2f57dbb42586045cf24a6a4d6d7178458905a804d77d0be26ff2f9L355-L357